model/operation.cc

```diff
@@ -135,7 +135,38 @@ namespace minsky
       ports.emplace_back
         (new Port(*this, Port::inputPort | (multiWire()? Port::multiWire: Port::noFlags)));
   }
-
+  
+  ClickType::Type OperationBase::clickType(float xx, float yy)
+  {
+    double fm=std::fmod(rotation(),360);
+    bool notflipped=(fm>-90 && fm<90) || fm>270 || fm<-270;
+    Rotate r(rotation()+(notflipped? 0: 180),0,0); // rotate into variable's frame of reference
+    double z=zoomFactor();
+    float dx=xx-x(), dy=yy-y(), w=iWidth()*z, h=iHeight()*z; 
+    // make sure resize handles can be grabbed at corners of coupled integral variable. for feature 94.
+    if (const IntOp* i=dynamic_cast<const IntOp*>(this))
+      if (i->coupled()) {    
+		  float dl=xx-i->x(), dr=xx-i->intVar->x(), wl=i->iWidth()*z, wr=i->intVar->iWidth()*z+i->intVarOffset;
+          if (((fabs(fabs(dl)-wl) < 0.5*portRadius*z) || (fabs(fabs(dr)-wr) < 0.5*portRadius*z)) &&
+          fabs(fabs(dy)-h) < 0.5*portRadius*z &&
+          (fabs(hypot(dl,dy)-hypot(wl,h)) < 0.5*portRadius*z) || (fabs(hypot(dr,dy)-hypot(wr,h)) < 0.5*portRadius*z))
+              return ClickType::onResize;		  
+	  } 
+    if (fabs(fabs(dx)-w) < portRadius*z &&
+        fabs(fabs(dy)-h) < portRadius*z &&
+        fabs(hypot(dx,dy)-hypot(w,h)) < portRadius*z)
+      return ClickType::onResize;  
+    return Item::clickType(xx,yy);
+  }
+  
+  float OperationBase::scaleFactor() const
+  {
+	float z=zoomFactor();
+    float l=OperationBase::l*z, r=OperationBase::r*z, 
+      h=OperationBase::h*z;
+    return std::max(1.0,std::min(static_cast<double>(iWidth())*z/(std::max(l,r)),static_cast<double>(iHeight())*z/h));  
```

Comment:
You could maintain everything in float precision by stating the constant as 1.0f